js/map/mapObject.mjs

```diff
@@ -171,21 +177,16 @@ MapIcon.prototype.draw = function(ctx, mouseLoc, currentSelection){
     }
 
     //Draw extra gate icons
-    if(this.cell.hive.classname == "location" && this.cell.name.includes("Oblivion Gate")){
-        let n = this.cell.name;
-        if(n.includes("Random") && randomGateCount >= 40){
+    if(this.cell.name.includes("Oblivion Gate")){
+        let n = this.cell.notes;
+        this.cell.notes.split(", ").forEach(note => {
+            if(icons["Overlay_" + note]){
+                ctx.drawImage(icons["Overlay_" + note], screenSpaceIconOrigin.x, screenSpaceIconOrigin.y, this.width(), this.height());
```

Comment:
once again, I don't like using displayed text in backend logic (sooner or later, it ends in PAIN when you try to modify something and can't because it changes some weird backend thing that relied on it), but I don't see a better option at the moment